packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -88,6 +88,13 @@ export interface BaseServiceProps {
    * @default false
    */
   readonly longArnEnabled?: boolean;
+
+  /**
+   * The launch type on which to run your service.
+   *
```

Comment:
Is it possible for this not to be defined? The @default documentation should describe what are the implications

Reply:
Yes, if the launchtype is not defined, as per: https://docs.aws.amazon.com/AmazonECS/latest/userguide/service_definition_parameters.html

launchType

    The launch type on which to run your service. Accepted values are FARGATE or EC2. If a launch type is not specified, EC2 is used by default. For more information, see Amazon ECS Launch Types.

Reply:
We could default this to EC2 in the BaseService? The only concern I'd have is if the behaviour changes in the future where Fargate becomes the default type for instance, the CDK will have to also make that change, and this could lead to inconsistencies between what customers expect, and what actually happens.

Reply:
Yea I share the same concern

Reply:
There is way for the service to change the default without breaking existing customers. A default value is part of the service API and practically cannot he changed. But I’d argue that in the CDK this actually needs to be a mandatory argument. I don’t see a “natural” default here from a user POV. I think the reason there’s a default here at the backend side is because this field was added only after Fargate was introduced, and they couldn’t break existing users that didn’t know about this property.

Reply:
I'm not opposed to this being a mandatory value, but if it is, should we have a default or expect the customer to always provide it?

Reply:
If it’s mandatory, there is no meaning for a default value 😝

Reply:
I'm changing it to a mandatory field